pkg/pods/factory.go

```diff
@@ -142,12 +141,14 @@ func PodFromPodHomeWithReqFile(node types.NodeName, home string, requireFile str
 	// Check if the pod home is namespaced by a UUID by splitting on a hyphen and
 	// checking the last part. If it parses as a UUID, pass it to newPodWithHome.
 	// Otherwise, pass a nil uniqueKey
-	homeParts := strings.Split(filepath.Base(home), "-")
+	podID := filepath.Base(home)
 
 	var uniqueKey types.PodUniqueKey
-	podUUID := uuid.Parse(homeParts[len(homeParts)-1])
-	if podUUID != nil {
-		uniqueKey = types.PodUniqueKey(podUUID.String())
+	if len(podID) > types.PodUUIDLength {
```

Comment:
what issue were you fixing here? it seems like if a pod ID is really long you'll mistake it for a uuid pod even if it isn't one

Reply:
that check is just so that it doesn't check for uuid in the directory name since it'll be too short.
Before the code was splitting the dir name by "-", so if you have a dir name of an app such as "pod-xxx-xxx-xx", homeParts will be ["pod","xxx","xxx","xx"] and homeParts[len(homeParts)-1] will only be the last part of the uuid. So I think the uuid check is invalid.

Reply:
See https://godoc.org/github.com/pborman/uuid#Parse

Reply:
oh i see, if it doesn't parse as a uuid you handle it fine, i'm still confused what was wrong before though

Reply:
It was not looking at the whole uuid in the dir name, i think the logic is wrong.

Reply:
ah I see, whoops

Reply:
also lol my comment 8 minutes ago was made before I refreshed, so it looks like i was responding to you but I didn't see your comments when i wrote that